index.php

```diff
@@ -35,7 +35,26 @@
 
 //Get sitepath
 $uri = rawurldecode($uri);
-$pattern = '(\/api\/|\/libs\/|\/js\/|\/css\/|\/images\/|\/files\/|\/admin\/|\/scripts\/|\/[^\/]*\.php|\/[^\/]*\.ico|\/auth_domain\/|\/auth_cookie\/|\/auth_token\/|\/login\/|\/utils\/|\/LEAF_test_endpoints\/)';
+//array of url segments that denote the end of the leaf sitepath
+$pathDelimiterArray = [
+    '/api/',
+    '/libs/',
+    '/js/',
+    '/css/',
+    '/images/',
+    '/files/',
+    '/admin/',
+    '/scripts/',
+    '/auth_domain/',
+    '/auth_cookie/',
+    '/auth_token/',
+    '/login/',
+    '/utils/',
+    '/LEAF_test_endpoints/',
+    '/[^/]*.php',
+    '/[^/]*.ico'
+];
+$pattern = '(' . addslashes(implode("|",$pathDelimiterArray)) . ')';
```

Comment:
Is there a way to do this string concatenation once in a constructor vs for every request?

I believe this code is on the hot code path